### PR TITLE
fix: e2e tests for deepspeed and pytorch geometric

### PR DESCRIPTION
**Pull request by liamcli**
## Description
Fix GPT-NeoX tests failing due to recent rebase on upstream that included API changes in the GPT-NeoX library.  
Improve pytorch geometric example to be more robust to data download failures by retrying a few times.

## Test Plan
Tested experiments in GCP cluster.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: examples/deepspeed/gpt_neox/gpt2_trial.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def __init__(self, context: DeepSpeedTrialContext) -> None:
             self.optimizer,
             self.lr_scheduler,
         ) = megatron_train.setup_model_and_optimizer(
-            neox_args=self.neox_args, inference=False, get_key_value=True
+            neox_args=self.neox_args
         )
```
Comment (rb-determined-ai) [inline examples/deepspeed/gpt_neox/gpt2_trial.py:61]: put this all on one line I think

File: examples/graphs/proteins_pytorch_geometric/model_def.py [hunk 2]
```diff
@@ -56,6 +57,20 @@ def forward(self, data):
...
+        try:
+            return TUDataset(
+                root=download_directory,
+                name=dataset_name
+            )
+        except Exception as e:
+            n_retries -= 1
+            if n_retries==0:
+                traceback.print_exc()
+    raise Exception("Failed to download data.")
```
Comment (rb-determined-ai) [inline examples/graphs/proteins_pytorch_geometric/model_def.py:71]: I'd prefer we don't raise a bare `Exception`, a `RuntimeError` or even a `ValueError` is preferable I think.

File: examples/graphs/proteins_pytorch_geometric/model_def.py [hunk 3]
```diff
@@ -56,6 +57,20 @@ def forward(self, data):
...
+    while n_retries > 0:
+        try:
+            return TUDataset(
+                root=download_directory,
+                name=dataset_name
+            )
+        except Exception as e:
+            n_retries -= 1
+            if n_retries==0:
+                traceback.print_exc()
```
Comment (rb-determined-ai) [inline examples/graphs/proteins_pytorch_geometric/model_def.py:70]: why not just keep the real exception?
```python
if n_retries == 0:
    raise
```

**Review by rb-determined-ai — APPROVED** [review]
